src/renderers/shared/server/ReactPartialRenderer.js

```diff
@@ -66,11 +56,21 @@ if (__DEV__) {
 
   var {ReactDebugCurrentFrame} = require('ReactGlobalSharedState');
   var currentDebugStack = null;
+  var currentDebugElementStack = null;
   var setCurrentDebugStack = function(stack) {
+    currentDebugElementStack = stack[stack.length - 1].debugElementStack;
+    // We are about to enter a new composite stack, reset the array.
+    currentDebugElementStack.length = 0;
     currentDebugStack = stack;
     ReactDebugCurrentFrame.getCurrentStack = getStackAddendum;
   };
+  var pushElementToDebugStack = function(element) {
+    if (currentDebugElementStack !== null) {
```

Comment:
I need this null check because this reset isn't reentrant so it might be reset. See follow up in https://github.com/facebook/react/issues/10188